tools/clang/lib/Headers/hlsl/vector_utils.h

```diff
@@ -0,0 +1,32 @@
+// Header for long vector APIs.
+
+#if ((__SHADER_TARGET_MAJOR > 6) ||                                            \
+     (__SHADER_TARGET_MAJOR == 6 && __SHADER_TARGET_MINOR >= 9)) &&            \
+    (__HLSL_VERSION >= 2021)
+
+#include <enable_if.h>
+
+namespace hlsl {
+
+// clang-format off
+template <int Off, int OSz, typename T, int ISz>
+typename hlsl::enable_if<Off + OSz <= ISz, vector<T, OSz> >::type
```

Comment:
So, it looks like the way we handle out of bounds would be via a compiler error that the function is undefined? Is that a pattern we use elsewhere? Just seemed a little weird to me, not saying I think it's wrong. Trying to see if I'm missing something.